browser/ui/brave_browser_command_controller.cc

```diff
@@ -78,6 +80,11 @@ void BraveBrowserCommandController::InitBraveCommandState() {
 #endif
   UpdateCommandForBraveAdblock();
   UpdateCommandForTor();
+  const base::CommandLine& command_line =
+      *base::CommandLine::ForCurrentProcess();
+  if (command_line.HasSwitch(switches::kEnableBraveSync)) {
+    UpdateCommandForBraveSync();
```

Comment:
we need to check the profile type (from `browser_->profile()->IsOffTheRecord()`)

Reply:
or alternatively we need to make sure this opens sync in a regular profile window and not in an incognito or guest profile